query/src/main/java/io/keen/client/java/MultiAnalysis.java

```diff
@@ -0,0 +1,172 @@
+
+
+    protected MultiAnalysis(Builder builder) {
+        super(builder);
+
+        this.analyses = builder.subAnalyses;
+    }
+
+    @Override
+    URL getRequestURL(RequestUrlBuilder urlBuilder, String projectId)
+            throws KeenQueryClientException {
+        return urlBuilder.getAnalysisUrl(projectId, KeenQueryConstants.MULTI_ANALYSIS);
+    }
+
+    @Override
+    Map<String, Object> constructRequestArgs() {
+        // First get the more general top-level args for this analysis.
+        Map<String, Object> multiAnalysisArgs = super.constructRequestArgs();
+
+        // validateParams() makes sure this.analyses is properly configured.
+        Map<String, Map<String, Object>> analysesArgs = new HashMap<String, Map<String, Object>>();
+
+        // Add each of the sub-analyses as "label": { ...sub-analysis args... }
+        for (SubAnalysis subAnalysis : this.analyses) {
+            Map<String, Object> subAnalysisArgs = subAnalysis.constructParameterRequestArgs();
+
+            analysesArgs.put(subAnalysis.getLabel(), subAnalysisArgs);
+        }
+
+        // Add the top-level "analyses" key
+        multiAnalysisArgs.put(KeenQueryConstants.ANALYSES, analysesArgs);
+
+        return multiAnalysisArgs;
+    }
+
+    @Override
+    protected void validateParams() {
+        super.validateParams();
+
+        if (null == this.analyses || this.analyses.isEmpty()) {
+            throw new IllegalArgumentException("MultiAnalysis requires that 'analyses' be set.");
+        }
+    }
+
+    /**
+     * This builder class helps configure the required and optional parameters that are appropriate
+     * for a MultiAnalysis.
+     */
+    public static class Builder extends CollectionAnalysis.Builder<Builder> {
+        // required
+        private Collection<SubAnalysis> subAnalyses;
+
+
+        /**
+         * Get the collection of sub-analyses for this multi-analysis.
+         *
+         * @return The sub-analyses.
+         */
+        public Collection<SubAnalysis> getSubAnalyses() { return this.subAnalyses; }
+
+
+        // Configurable properties added by this builder.
+
+        /**
+         * Set the collection of sub-analyses. Existing sub-analyses will be discarded.
+         *
+         * @param subAnalyses The new analysis arguments to replace the discarded ones, if any.
+         */
+        public void setSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            this.subAnalyses = null;
+
+            // Client code may just be clearing all the analyses.
+            if (null != subAnalyses) {
+                withSubAnalyses(subAnalyses);
+            }
+        }
+
+        /**
+         * Adds a collection of sub-analyses.
+         *
+         * @param subAnalyses The new analysis arguments to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
```

Comment:
I'm with @baumatron on this one, and he and I of course discussed this a bit before going this route. It seemed more natural and more predictable to be able to interlace calls to ```withStep()``` and ```withSteps()``` at will, in any order, and any number of times with identical behavior. To me that's what with*() means for lists of things. For single properties, I think with*() should impose a one-time-only protocol, and we did that too, because why is your client code trying to with*() the collection name twice instead of calling set*()? But with the current design, withSteps() and withStep() can only be used either mutually-exclusively or in a very precise order to make any sense at all, which doesn't seem in the spirit of fluent APIs. To me, set*() is another beast entirely. Or are we thinking about that strangely?

Anyway, here since it's new code, we have the option to do it the way we all decide is best, or strictly adhere to how it used to be done for filters (those may be the same or different) for consistency, with the plan to potentially change it across the SDK in a major revision. Thoughts?